UI/Components/ServerComponent.cs

```diff
@@ -387,21 +404,30 @@ private void ProcessMessage(String message, Connection clientConnection)
                             title = options[1];
                         }
 
-                        State.Run[index].Name = title;
-                        State.Run.HasChanged = true;
+                        if (index >= 0)
+                        {
+                            State.Run[index].Name = title;
+                            State.Run.HasChanged = true;
+                        }
+
                         break;
                     }
                     default:
                     {
-                        // perhaps an error should be returned for an unrecognized message?
-                        break;
+                        throw new Exception("Unrecognized message");
```

Comment:
Would it make sense to include the command in the error message? Something like:
```suggestion
                        throw new Exception($"Unrecognized command: {command}");
```

Reply:
Yeah that's a good idea